internal/appsec/waf.go

```diff
@@ -95,9 +95,35 @@ func registerWAF(rules []byte, timeout time.Duration, limiter Limiter, obfCfg *O
 // newWAFEventListener returns the WAF event listener to register in order to enable it.
 func newHTTPWAFEventListener(handle *waf.Handle, addresses []string, timeout time.Duration, limiter Limiter) dyngo.EventListener {
 	var monitorRulesOnce sync.Once // per instantiation
+	actionHandler := httpsec.NewActionsHandler()
 
 	return httpsec.OnHandlerOperationStart(func(op *httpsec.Operation, args httpsec.HandlerOperationArgs) {
 		var body interface{}
+		wafCtx := waf.NewContext(handle)
+		if wafCtx == nil {
+			// The WAF event listener got concurrently released
+			return
+		}
+
+		values := map[string]interface{}{}
+		for _, addr := range addresses {
+			if addr == httpClientIP && args.ClientIP.IsValid() {
+				values[httpClientIP] = args.ClientIP.String()
+			}
+		}
```

Comment:
The for-loop is not necessary in this simplified ip-blocking-only case:
```suggestion
		if _, exists := addresses[httpClientIP]; exists && args.ClientIP.IsValid() {
			values[httpClientIP] = args.ClientIP.String()
		}
		// TODO: suspicious request blocking by moving here all the addresses available when the request begins
```

Reply:
I was to quick to commit this. Actually `addresses` is a slice here, not a map